back/app/services/idea_custom_fields_service.rb

```diff
@@ -141,6 +141,58 @@ def check_form_structure(fields, errors)
     end
   end
 
+  def duplicate_all_fields
+    fields = all_fields
+    logic_id_map = { survey_end: 'survey_end' }
+    copied_fields = fields.map do |field|
+      # Duplicate fields with a new id
```

Comment:
```suggestion
      # Duplicate fields and add a new id
```
Maybe clearer? Original could be read as meaning; "only copy fields that have a new ID"_